### ISSUE TITLE
Adding docs to TagValue enumeration values

### ISSUE THREAD

**Issue by nickbar01234** [palantir/metric-schema#1016]
Some enumeration values are not self-explanatory and requires additional documentation to understand metric definitions. Currently, it's not possible to do so because `metrics-schema-api.yml` defines `TagValue` as

```yml
TagValue:
  fields:
    value: string
```

My work-around so far is to add the documentation for enumeration values in `docs` for `TagDefinition`, but this feels a bit clunky. Would it be possible / make-sense to re-define the schema definition to the following and update the markdown generator to output documentations for enumeration values?

```yml
TagValue:
  fields:
    value: string
    docs: optional<Documentation>
```

**Comment by carterkozak** [comment]
That sounds like a great improvement, I've also wanted to document individual tag values! Would you like to contribute this feature?

In addition to the markdown generator, this would be helpful to include as javadoc on our generated tag value enumerations!

**Comment by nickbar01234** [comment]
Yup, I can take a stab at implementing it!

**Comment by nickbar01234** [comment]
Hey @carterkozak, following up on your comment about Javadoc. The current [code generation](https://github.com/palantir/metric-schema/blob/develop/metric-schema-java/src/main/java/com/palantir/metric/schema/UtilityGenerator.java#L71) only builds `enum class` when there are at least 2 enumeration values. Otherwise, the tag value is statically defined in the output code, which makes it hard to add Javadoc to a specific part of the output code. Do you think it makes sense to output an `enum class` if there is at least 1 tag value? 

Also, I'm not very familiar with Javadoc, I would love to hear your thoughts on what I'm planning to do 😁. I imagine the JavaDoc for this metric-schema

```yml
metric:
  tags:
    - name: enum
    - values:
         -  FOO
         -  value: BAR
             docs: Some documentation
```

would yield

```java
// FOO doesn't have @see because there's no documentations
/**
 * @see Metric_Enum#BAR
 */
public enum Metric_Enum {
    FOO("foo")
    /**
     * Some documentation
     */
    BAR("bar")
    ...
}
```

**Comment by nickbar01234** [comment]
Resolved in #1021.
